src/resolve-dependency.ts

```diff
@@ -9,13 +9,14 @@ export default function resolveDependency (specifier: string, parent: string, jo
   if (isAbsolute(specifier) || specifier === '.' || specifier === '..' || specifier.startsWith('./') || specifier.startsWith('../')) {
     const trailingSlash = specifier.endsWith('/');
     resolved = resolvePath(resolve(parent, '..', specifier) + (trailingSlash ? '/' : ''), parent, job);
-  }
-  else {
+  } else if (typeof job.resolve === 'function') {
+    resolved = job.resolve(specifier, parent);
+  } else {
     resolved = resolvePackage(specifier, parent, job, cjsResolve);
   }
 
   if (Array.isArray(resolved)) {
-    return resolved.map(resolved => job.realpath(resolved, parent));
+    return resolved.filter(Boolean).map(resolved => job.realpath(resolved, parent));
```

Comment:
Can you clarify why this is added here?

Reply:
because now we're dealing with outside input (that might not be typed). this is just a small assurance that the array has truthy things in  it before using them in `realpath()`

Reply:
Seems like that should be validated earlier?